java-client/src/main/java/org/opensearch/client/opensearch/indices/IndexSettings.java

```diff
@@ -198,12 +198,18 @@ public class IndexSettings implements JsonpSerializable {
 	@Nullable
 	private final Integer maxSlicesPerScroll;
 
+	@Nullable
+	private final Translog translog;
+
 	@Nullable
 	private final String translogDurability;
```

Comment:
It seems like the `translogDurability`, `translogFlushThresholdSize` and `translogSyncInterval` should be removed in favor of `translog`?

Reply:
Yeah, such a duplication seems ugly enough. Thank you for the feedback!

I have added a second commit, please check it out and verify that I have got the public API aspects right. I wrote the main ideas in the commit message.

Additionally, I think we will have to change serialization from
```
{
    "translog.durability": "async"
}
```
to
```
{
    "translog": {
        "durability": "async"
    }
}
```
(second commit supports both to ensure backward compatibility). As a result, only the latter form will be recognized by the deserializer. How should we mark the former one as deprecated?

Reply:
Thank you!